### PR TITLE
feat(mobile): add device detection and screen orientation locking to portrait

### PR DESCRIPTION
**Pull request by lgmarchi**
### What this PR does 📖

> [!IMPORTANT]
> Screen orientation lock was added, because app is not working really good yet in landscape orientation 

This pull request includes changes to add mobile device detection and screen orientation locking functionality. The main changes involve importing new dependencies, adding a utility function to detect mobile platforms, and locking the screen orientation for specific devices.

### Mobile device detection and screen orientation locking:

* [`src/lib/utils/Mobile.ts`](diffhunk://#diff-8ff4a5d9edf728fc9c713852815e46bc99e4cc51dcc799a244c8d11abf275828R1-R10): Added a new function `isAndroidOriOS` to detect if the device is running on Android or iOS.
* [`src/routes/+layout.svelte`](diffhunk://#diff-cc69d6a97e4f62578028d872b8d5032f2786fe1eaa65735429286b375234168dR30-R31): Imported `ScreenOrientation` and `isAndroidOriOS` to handle screen orientation locking.
* [`src/routes/+layout.svelte`](diffhunk://#diff-cc69d6a97e4f62578028d872b8d5032f2786fe1eaa65735429286b375234168dR269-R281): Added the `lockOrientation` function to lock the screen orientation to portrait mode and integrated it with the `onMount` lifecycle method to execute on supported devices.

### Which issue(s) this PR fixes 🔨

-   Resolve #

### CODE REVIEW

File: ios/App/Podfile.lock [hunk 1]
```diff
@@ -23,18 +29,24 @@ EXTERNAL SOURCES:
...
+  CapacitorDevice: 1a215717f0b5061503b21a03508b0ec458a57d78
+  CapacitorFilesystem: c832a3f6d4870c3872688e782ae8e33665e6ecbf
+  CapacitorScreenOrientation: 3bb823f5d265190301cdc5d58a568a287d98972a
   CapacitorSplashScreen: 250df9ef8014fac5c7c1fd231f0f8b1d8f0b5624
 
-PODFILE CHECKSUM: 46011c4258f4c76d658cc7d9721149c3759c52d7
+PODFILE CHECKSUM: 99b0f84cf09d9f31eaad5636c4a6b9c8ca734a2c
 
-COCOAPODS: 1.16.1
+COCOAPODS: 1.15.2
```
Comment (stavares843) [inline ios/App/Podfile.lock]: this is being downgraded, i believe you need to update your pods locally

**Review by stavares843 — COMMENTED** [review]
@lgmarchi we don't need any of that to force the orientations 


just this updating from

Captura de ecrã 2024-11-28, às 22 10 47




to this

Captura de ecrã 2024-11-28, às 22 27 35

and will force the portrait even if you are on landscape

Captura de ecrã 2024-11-28, às 22 28 09

**Review by stavares843 — CHANGES_REQUESTED** [review]
added comments above

**Comment by stavares843** [comment]
i mean if you want we can add those plugins and the utility function you added regardless @lgmarchi 

was just explaining that it was not needed to achieve what you mentioned in this case, to block the landscape orientation

**Comment by lgmarchi** [comment]
> @lgmarchi we don't need any of that to force the orientations
> 
> just this updating from
> 

This code can work for Android devices as well. And it is easier to just remove when need to give maintenance on it,

> i mean if you want we can add those plugins and the utility function you added regardless @lgmarchi
> 
> was just explaining that it was not needed to achieve what you mentioned in this case, to block the landscape orientation

Both approaches work, but I chose this method because it keeps the code consistent across iOS and Android, making it easier to manage. It allows dynamic control of orientation—like locking portrait for most pages but enabling landscape for specific ones when needed. It’s also simpler to update or remove during testing compared to editing platform-specific files like `Info.plist` or `AndroidManifest.xml`.

**Comment by stavares843** [comment]
Again, not against merging the PR with the plugins and the utility function though.

>  It allows dynamic control of orientation—like locking portrait for most pages but enabling landscape for specific ones when needed.

but seems we aren't using that on the PR - can you add a example for when we use that in the future then?

Can you update the pods? They are downgrading on this PR.

**Comment by luisecm** [comment]
Minor comment, we should use our logger instead of console.error

**Comment by lgmarchi** [comment]
> Member


For example, when we have landscape videos or images, users naturally want to view them in landscape orientation. During group calls, people might prefer landscape mode too, since it offers a wider view. Even in the storage section, being able to use landscape orientation is super helpful—it lets users see their files and the file tree at the same time.

**Comment by stavares843** [comment]
Yes I get that I meant a example in code

**Comment by stavares843** [comment]
also, if I wanted to speak with AI I would go to chatgpt

**Comment by lgmarchi** [comment]
> Yes I get that I meant a example in code

You asked for a future example—`could you kindly provide an example of when we might use that in the future?` I've provided my answer to clarify.

**Comment by stavares843** [comment]
no i did not said that lol

i said

> can you add a example for when we use that in the future then?


i mean in code - not in theory obviously

if you are adding this code, we could also add the theory of what pages to hide right? then anyone knows how to add/delete one in the future

**Comment by luisecm** [comment]
@lgmarchi Hello Lucas, I see the same issue again on this branch - Install Banner is not shown on web. Could you please address this bug? Seems like every time we touch files for some strange reason the install banner is being affected


https://github.com/user-attachments/assets/0f10e511-36ef-4a8c-a3c8-3491d77ce846

**Review by stavares843 — CHANGES_REQUESTED** [review]
luis comment needs fixing

**Comment by stavares843** [comment]
we will retest with the latest PR that was merged regarding luis comment

**Comment by phillsatellite** [comment]
Hey @lgmarchi so during testing I was able to turn the device horizontally and rest of the app changed orientation as well. The goal of this pr is to block this correct? 

Screenshot 2024-12-04 at 11 50 44 AM

**Comment by lgmarchi** [comment]
> Hey @lgmarchi so during testing I was able to turn the device horizontally and rest of the app changed orientation as well. The goal of this pr is to block this correct?
> 
> Screenshot 2024-12-04 at 11 50 44 AM

Yes, it should not happen. 

Did you do npm install before? Because there is a new lib that need to be used. And you can not change branch and test. 

You need to change branch, and run code to run app again from beginning.

**Comment by phillsatellite** [comment]
> > Hey @lgmarchi so during testing I was able to turn the device horizontally and rest of the app changed orientation as well. The goal of this pr is to block this correct?
> > Screenshot 2024-12-04 at 11 50 44 AM
> 
> Yes, it should not happen.
> 
> Did you do npm install before? Because there is a new lib that need to be used. And you can not change branch and test.
> 
> You need to change branch, and run code to run app again from beginning.

Did do all of these steps. Sara had tried a first time and it worked but now her second time trying it isnt working. Could be from when the conflicts got fixed?

**Comment by lgmarchi** [comment]
> > > Hey @lgmarchi so during testing I was able to turn the device horizontally and rest of the app changed orientation as well. The goal of this pr is to block this correct?
> > > Screenshot 2024-12-04 at 11 50 44 AM
> > 
> > 
> > Yes, it should not happen.
> > Did you do npm install before? Because there is a new lib that need to be used. And you can not change branch and test.
> > You need to change branch, and run code to run app again from beginning.
> 
> Did do all of these steps. Sara had tried a first time and it worked but now her second time trying it isnt working. Could be from when the conflicts got fixed?

Yes, it was. I saw code, conflict mix some code.

**Review by stavares843 — APPROVED** [review]

**Review by phillsatellite — APPROVED** [review]

**Pull request merged**